analyses/explore-spotify-variation.Rmd

```diff
@@ -56,3 +63,61 @@ ggplot(pca_df) +
              alpha = 0.5) + 
   theme_bw()
 ```
+
+## UMAP
+
+```{r}
+# Calculate UMAP itself with default parameters
+umap_result <- umap(pca$x)
+```
+
+```{r}
+# Extract UMAP1 and UMAP2 into a data frame
+umap_df <- data.frame(umap_result$layout) |>
+  # rename X1, X2 --> UMAP1, UMAP2
+  dplyr::rename_all(.funs = gsub,
+                    pattern = "^X",
+                    replacement = "UMAP")
+```
+
+```{r}
+# plot the UMAP:
+ggplot(umap_df) +
```

Comment:
For these UMAPs, can we set a different plot theme? Since axes aren't meaningful in UMAPs, we probably don't want to emphasize axis ticks and labels, just the title (UMAP1 and UMAP2). Adding a theme here and to other UMAPs (or set a theme before UMAPs get plotted? Up to you!) would help.